src/main/java/org/terasology/accelerationblocks/AccelerationSystem.java

```diff
@@ -0,0 +1,150 @@
+import org.terasology.math.geom.Quat4f;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.registry.In;
+import org.terasology.world.BlockEntityRegistry;
+import org.terasology.world.WorldProvider;
+import org.terasology.world.block.Block;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class AccelerationSystem extends BaseComponentSystem implements UpdateSubscriberSystem {
+    private static final Logger logger = LoggerFactory.getLogger(AccelerationSystem.class);
+
+    @In
+    private BlockEntityRegistry blockEntityProvider;
+
+    @In
+    private Time time;
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private WorldProvider worldProvider;
+
+    @Override
+    public void update(float delta) {
+        for (EntityRef entity : entityManager.getEntitiesWith(AccelerationComponent.class, LocationComponent.class)) {
+            AccelerationComponent acceleration = entity.getComponent(AccelerationComponent.class);
+            LocationComponent location = entity.getComponent(LocationComponent.class);
+
+            applyImpulse(acceleration, entity);
+        }
+    }
+
+    private void applyImpulse(AccelerationComponent acceleration, EntityRef entity) {
+        Vector3f impulse = new Vector3f(acceleration.velocity);
+        entity.send(new CharacterImpulseEvent(impulse));
+    }
+
+    @ReceiveEvent
+    public void onEnterBlock(OnEnterBlockEvent event, EntityRef entity) {
+        //ignores "flying" block
+        if (isAtHeadLevel(event.getCharacterRelativePosition(), entity)) {
+            return;
+        }
+
+
+        Block block = event.getNewBlock();
+        if (blockIsAccelerating(block)) {
+            AccelerationBlockComponent blockAcceleration = block.getEntity().getComponent(AccelerationBlockComponent.class);
+            AccelerationComponent acceleration = entity.getComponent(AccelerationComponent.class);
+
+            if (acceleration == null) {
+                acceleration = new AccelerationComponent();
+                initAcceleration(acceleration, blockAcceleration, block.getDirection());
+                entity.addComponent(acceleration);
+            } else {
+                initAcceleration(acceleration, blockAcceleration, block.getDirection());
+                entity.saveComponent(acceleration);
+            }
+        } else {
+            //check if it was accelerated before and removes acceleration component
+            AccelerationComponent accelerationOld = entity.getComponent(AccelerationComponent.class);
+
+            if (accelerationOld != null) {
+                entity.removeComponent(AccelerationComponent.class);
+            }
+        }
+    }
+
+    private void initAcceleration(AccelerationComponent acceleration, AccelerationBlockComponent blockAcceleration, Side side) {
+        Vector3f velocity = new Vector3f(blockAcceleration.velocity);
+        if(!blockAcceleration.ignoreBlockDirection){
```

Comment:
Tiny code style fix: single space between `) {`

There are a few others like that, try to hook up and run [Checkstyle](https://github.com/MovingBlocks/Terasology/wiki/Checkstyle) to find and fix them all :-)